keps/sig-cli/3655-kubectl-translation-improvement/README.md

```diff
@@ -0,0 +1,1226 @@
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+1. Adding or Removing locales from the supported list. 
+   * While there may be merit to the argument that the list of supported locales should be changed, it is beyond the scope of this KEP and should be handled separately.
+2. Rewriting kubectl's i18n code or making significant changes to how it works.
+   * There may be some minor changes needed, but the goal of this KEP is to improve on what we already have in place.
+3. Translating text that is generated from the API, controllers, plugins, or other sources outside kubectl. 
+   * While some translation might be possible from these sources, it can be complex due to their dynamic nature and is out of scope for this KEP.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### 1. Make it simple for people to update translations.
+
+To update translations today, you first need to install some prerequisites and run a script in the hack directory before you can do what you really want to do, which is to translate text.
+
+This shouldn't be required.
+
+This KEP will define how the translation files can be kept up-to-date with new and updated text placeholders from the code, so that translation contributors do not need to perform this unrelated task and the process becomes simpler for them.
+
+### 2. Identify and empower contributors who will be approvers and reviewers for each locale.
+
+Each locale should have trusted contributors who can function as reviewers and approvers for its translations. We can manage this through the use of directory-specific [OWNERS](https://www.kubernetes.dev/docs/guide/owners/) files.
+
+This KEP will specify who should be added to the OWNERS file for each locale.
+
+### 3. Improve documentation.
+
+Currently, the documentation on how to contribute as a translator exists within the kubectl source code in a README.md. This document is not easy to discover and not friendly to first time contributors.
+
+This KEP will call for the documentation to be updated and for it to be linked from a new, more prominent location where contributors can learn how to translate kubectl's text.
+
+### 4. Ensure all strings in code support translations.
+
+There are two factors contributing to the problem of incomplete translations:
+1. Text that is available to be translated, but has not been.
+2. Text that is not available for translation because it has not yet been wrapped in `i18n.T()`.
+
+So far, this KEP has focused on the first aspect, but the second aspect must also be addressed.
```

Comment:
You mean this part? 

> As part of the effort to implement this KEP, we will wrap all translatable text in `i18n.T()` where possible within kubectl.

I'm open to any ideas.  My only concern is whether translations will be good enough after all the work is done, so I want to make sure we address all the deficiencies and lack of `i18n.T()` usage is one of them.

I envision this KEP kicking off several separate, but related issues, and this would be one of them.